### DISCUSSION TITLE
How secure is my database without RLS and can I use RLS without user accounts?

### DISCUSSION THREAD

**Discussion by cwilharm** [supabase/supabase#9894] [Questions] [answered]
I wonder how secure my database is without RLS. I do not use user accounts, so Auth is unfortunately uninteresting for me... Can someone help me with this question?

**Comment by burggraf** [comment, marked as answer]
It is safe to not use RLS only if:
1.  You do not use user accounts
2.  You only access the database from a secure server environment
3.  You do not use either your anon key or your service key anywhere in your client code (and you keep those keys secure.)

However, if you are using swift or any other language that eventually can be reverse-engineered at the client side, then you should use RLS, as your key can become compromised.  In general, any time  a client is accessing the database directly you should use an RLS policy.

Mark

**Reply by cwilharm** [reply]
Ok thanks a lot! Is there a way to not use user accounts in my Swift app?
